libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/UnpackUpdateTask.java

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+
+import java.io.File;
+
+/**
+ * This class is an {@link AsyncTask} which monitors the unpacking of a TAR file and updates a
+ * listener so that the view can show the unpacking progress. It monitors the unpacking by
+ * periodically checking the file size, because as it's unpacked, the file size will decrease.
+ */
+public class UnpackUpdateTask extends AsyncTask<File, Long, File> {
+  private ProgressUpdateListener progressUpdateListener;
+
+  /**
+   * Creates a new UnpackUpdateTask to update the view via a passed {@link ProgressUpdateListener}.
+   *
+   * @param progressUpdateListener listener to update
+   */
+  UnpackUpdateTask(@NonNull ProgressUpdateListener progressUpdateListener) {
+    this.progressUpdateListener = progressUpdateListener;
+  }
+
+  @Override
+  protected File doInBackground(File... files) {
+    File tar = files[0];
+    long size = tar.length();
+
+    while (tar.length() > 0) {
+      publishProgress((((tar.length() / size)) * 100));
```

Comment:
I noticed that the published progress here is either `0` or `100` - was this expected behavior?

![screen shot 2018-11-28 at 3 50 56 pm](https://user-images.githubusercontent.com/8434572/49181943-d6a14480-f326-11e8-8b69-443d4f6832ae.png)

@Guardiola31337 said he had some ideas about a buffer to monitor this?  @kevinkreiser any ideas what the issue is with monitoring the tar unpacking?  Thanks!